### DISCUSSION TITLE
Setting up Mastodon Server on Intranet using IP Address instead of Domain Name

### DISCUSSION THREAD

**Discussion by Surprise-Jing** [mastodon/mastodon#27418] [Q&A]
Is it possible to deploy Mastodon within an intranet environment using the server's IP address instead of a domain name. I'm seeking guidance on how to perform this installation.

**Comment by mjankowski** [comment]
It is POSSIBLE (read about LOCAL_DOMAIN and WEB_DOMAIN interaction) ... but advised against. Can you use whatever DNS you have on the local network to point some non-external hostname at the instance?

**Reply by roke-julian-lockhart** [reply]
> It is POSSIBLE (read about LOCAL_DOMAIN and WEB_DOMAIN interaction) ... but advised against.

Why is it advised against, @mjankowski?

**Reply by mjankowski** [reply]
A few reasons:

- To the extent that the hostname forms a portion of a human-readable identifier, it seems more usable/memorable to involve a hostname instead of an IP Address alone
- To the extent that you want the server to be somewhat long-lived, it seems generally easier to ensure that hostnames survive any network changes more than IPs
- Related to last point, for purposes of other instances connecting successfully over time, seems more likely to happen w/ a hostname

Vaguely related - https://github.com/mastodon/mastodon/pull/32526 - might lead to a doc PR on how to do this if you need to.
